stack/stack.go

```diff
@@ -55,6 +55,10 @@ type (
 		// before is a list of stack paths that must run after this stack.
 		before []string
 
+		// implicitBefore is a map of paths that must run before this stack but
+		// they were implicitly computed from the project hierarchy.
+		implicitBefore map[string]struct{}
```

Comment:
why do we need a separate implicit before instead of just adding on the stacks before ? the errors in general are the samy anyway no ? like cycles etc ? (I imagine Im missing something, so just curious).

Reply:
hmm I feel you...but I find it a little odd that we have extra logic that is unused, nothing on the code uses this difference between implicit/normal before, as soon as I saw it I kept looking for where it is used..and the fact that it is not used feels like unnecessary complexity (extra abstractions/code). But if you are confident it will be useful in the future for improved diagnostics it is all good :+1: 

On directly mutating before, we can have a method to add the Befores (instead of direct mutation) , because in the end mutation happens anyway in the sense that calling Before() returns the mutated (merged) befores, so it already has some mutation complexity but in a way that seems harder to debug, like someone debugging this may log the before stacks and forget about implicitBefore, inspecting before VS Before() will produce different results, etc. I would be more prone to just modeling on the stack a way to clearly add before stacks (but not necessarily exposing the slice directly) and then use that on every place that needs before stacks to be added (the implicit and also when loading config, etc).

But it is just an idea and my initial perception, feel free to disregard.